doc/releases/changelog-dev.md

```diff
@@ -4,6 +4,11 @@
 
 <h3>New features since last release</h3>
 
+* New transforms called :func:`~.transforms.replace_relative_phase_toffoli` and 
+  :func:`~.transforms.replace_controlled_iX_gate` have been added to implement passes that make use
+  of equivalencies to compile certain patterns to efficient Clifford+T equivalents.
+  [(#7748)](https://github.com/PennyLaneAI/pennylane/pull/7748)
```

Comment:
I don't love the names, but I'm not sure I like any alternatives either 😂. I think this just speaks to the fact that we need better pattern matching capabilities. We have `qml.pattern_matching` but it requires a DAG as input, and I understand the desire to have "stock" pattern matches regardless.

Let's just keep the names as is for now because I don't have a better idea here, and it would require us to rethink pattern matching overall (something we have discovery for this quarter!)